publisher/publisher.go

```diff
@@ -111,7 +113,16 @@ func (pub *publisher) Publish(contentType string, content []byte, config map[str
 		tags := map[string]string{}
 
 		// at least one tag is required by KairosDB
-		tags["hostname"] = metric.Source()
+		if hostname, ok := metric.Tags()[core.STD_TAG_PLUGIN_RUNNING_ON]; ok {
+			tags["hostname"] = hostname
+		} else {
+			hostname, err := os.Hostname()
+			if err != nil {
+				tags["hostname"] = "localhost"
```

Comment:
I agree. So I think you should be able to therefore take out lines 116-125 because the requirement of 1 tag is satisfied by the `core.STD_TAG_PLUGIN_RUNNING_ON` tag and make line 137 `Tags: metric.Tags()`